packages/vscode/src/telemetryReporter.ts

```diff
@@ -24,9 +32,27 @@ export default class TelemetryReporter {
   }
 
   private updateUserOptIn() {
-    const config = workspace.getConfiguration(TelemetryReporter.TELEMETRY_CONFIG_ID)
-    if (this.userOptIn !== config.get<boolean>(TelemetryReporter.TELEMETRY_CONFIG_ENABLED_ID, true)) {
-      this.userOptIn = config.get<boolean>(TelemetryReporter.TELEMETRY_CONFIG_ENABLED_ID, true)
+    const telemetrySettings = workspace.getConfiguration(TelemetryReporter.TELEMETRY_SECTION_ID)
+    const isTelemetryEnabled = telemetrySettings.get<boolean>(TelemetryReporter.TELEMETRY_OLD_SETTING_ID)
+    // Only available since https://code.visualstudio.com/updates/v1_61#_telemetry-settings
+    const telemetryLevel = telemetrySettings.get<string>(TelemetryReporter.TELEMETRY_SETTING_ID) as TelemetryLevel
+
+    // `enableTelemetry` is either true or false (default = true). Deprecated since https://code.visualstudio.com/updates/v1_61#_telemetry-settings
+    // It is replaced by `telemetryLevel`, only available since v1.61 (default = 'all')
+    // https://code.visualstudio.com/docs/getstarted/telemetry
+    // To enable Telemetry:
+    // We check that
+    // `enableTelemetry` is true and `telemetryLevel` falsy -> enabled
+    // `enableTelemetry` is true and `telemetryLevel` set to 'all' -> enabled
+    // anything else falls back to disabled.
+    const isTelemetryEnabledWithOldSetting = isTelemetryEnabled && !telemetryLevel
+    const isTelemetryEnabledWithNewSetting = isTelemetryEnabled && telemetryLevel && telemetryLevel === 'all'
```

Comment:
What we currently send is the version used and metadata only, so it would be a stretch to enable it on "crash" or "error" since we don't collect any information about a potential crash/error anyway at the moment.

Also, the telemetry event is triggered only when the extension is activated  = when opening a `schema.prisma` file.